dll/win32/lpk/lpk.c

```diff
@@ -132,28 +149,38 @@ LpkGetCharacterPlacement(
     lpResults->nGlyphs = (UINT)cGlyphs;
 
     if (lpResults->lpGlyphs)
-        wcscpy(lpResults->lpGlyphs, lpGlyphs);
-
-    if (lpResults->lpDx && !(dwFlags & GCP_GLYPHSHAPE))
     {
-        int c;
-        for (i = 0; i < nSet; i++)
-        {
-            if (GetCharWidth32W(hdc, lpResults->lpOutString[i], lpResults->lpOutString[i], &c))
-                lpResults->lpDx[i] = c;
-        }
+        if (lpGlyphs)
+            StringCchCopyW(lpResults->lpGlyphs, cGlyphs, lpGlyphs);
+
+        else if (lpResults->lpOutString)
+            GetGlyphIndicesW(hdc, lpResults->lpOutString, nSet, lpResults->lpGlyphs, 0);
     }
 
-    /* If glyph shaping was requested */
-    else if (lpResults->lpDx && (dwFlags & GCP_GLYPHSHAPE))
+    if (lpResults->lpDx)
     {
-        int c;
+        /* If glyph shaping was requested */
+        if (dwFlags & GCP_GLYPHSHAPE)
+        {
+            int c;
```

Comment:
Move this `int c;` just above (place it as the first thing in the `if (lpResults->lpDx)`).
This allows avoiding yet again an `int c;` declaration in the 'else' block.